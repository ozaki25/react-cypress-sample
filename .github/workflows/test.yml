name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
