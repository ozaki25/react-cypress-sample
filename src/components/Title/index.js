import React from 'react';
import styled from 'styled-components';

const H1 = styled.h1`
  font-size: 3em;
  padding: 0.3em;
  background: #e3f2fd;
  border-bottom: solid 5px #1565c0;
  margin: 0 0 15px;
`;

function Title(props) {
  return <H1 {...props} />;
}

export default Title;
