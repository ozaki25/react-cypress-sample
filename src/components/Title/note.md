- 見出しのタイトルに使うコンポーネントです