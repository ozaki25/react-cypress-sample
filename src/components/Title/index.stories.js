import React from 'react';
import Title from '.';
import note from './note.md';

export default {
  title: 'Title',
  parameters: {
    info: {
      text: note,
    },
  },
};

export const nomal = () => <Title>タイトル</Title>;
