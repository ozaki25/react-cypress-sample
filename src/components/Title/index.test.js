import React from 'react';
import { snapshot } from '../../test/helper';
import Title from '.';

snapshot('Title', <Title>タイトル</Title>);
