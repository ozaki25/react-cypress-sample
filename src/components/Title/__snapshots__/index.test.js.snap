// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Title 1`] = `
<h1
  className="sc-bdVaJa jXsmnv"
>
  タイトル
</h1>
`;
